data/npc/scripts/bank.lua

```diff
@@ -233,6 +233,12 @@ local function creatureSayCallback(cid, type, msg)
 			end
 		end
 	elseif npcHandler.topic[cid] == topicList.TRANSFER_PLAYER_GOLD then
+		local currencyValue = tonumber(msg)
```

Comment:
Better off making sure `getMoneyCount` returns a numeric value that is acceptable rather than adding checks like this (which may be required in other places as well).

Reply:
In https://github.com/otland/forgottenserver/blob/4929928966abdfbdb786d5ed297034785d08edf8/data/npc/scripts/bank.lua#L141 change ` if isValidMoney(count[cid]) then` to `if count[cid] ~= -1 then` ?

Reply:
`				-- Immediate topicList.TRANSFER_PLAYER_GOLD simulation
				count[cid] = getMoneyCount(parts[2])
				if count[cid] ~= -1 then
					if player:getBankBalance() < count[cid] then
						npcHandler:say("There is not enough gold in your account.", cid)
						npcHandler.topic[cid] = topicList.NONE
						return true
					end
					npcHandler:say("Who would you like transfer " .. count[cid] .. " gold to?", cid)
					npcHandler.topic[cid] = topicList.TRANSFER_PLAYER_WHO
				else
					npcHandler:say("There is not enough gold in your account.", cid)
					npcHandler.topic[cid] = topicList.NONE
				end`